centreon/src/Core/Notification/Application/UseCase/PartialUpdateNotification/PartialUpdateNotification.php

```diff
@@ -0,0 +1,165 @@
+     * @param int $notificationId
+     */
+    public function __invoke(
+        PartialUpdateNotificationRequest $request,
+        PartialUpdateNotificationPresenterInterface $presenter,
+        int $notificationId
+    ): void {
+        try {
+            if ($this->contactCanExecuteUseCase()) {
+                $response = $this->partiallyUpdateNotification($request, $notificationId);
+            } else {
+                $this->error(
+                    "User doesn't have sufficient rights to delete notifications",
+                    ['user_id' => $this->contact->getId()]
+                );
+                $response = new ForbiddenResponse(NotificationException::partialUpdateNotAllowed()->getMessage());
+            }
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+            $response = new ErrorResponse(NotificationException::errorWhilePartiallyUpdatingObject());
+        }
+
+        $presenter->presentResponse($response);
+    }
+
+    /**
+     * @return bool
+     */
+    private function contactCanExecuteUseCase(): bool
+    {
+        return $this->contact->hasTopologyRole(Contact::ROLE_CONFIGURATION_NOTIFICATIONS_READ_WRITE);
+    }
+
+    /**
+     * @param PartialUpdateNotificationRequest $request
+     * @param int $notificationId
+     *
+     * @throws \Throwable
+     *
+     * @return ResponseStatusInterface
+     */
+    private function partiallyUpdateNotification(
+        PartialUpdateNotificationRequest $request,
+        int $notificationId
+    ): ResponseStatusInterface {
+        if (! ($notification = $this->readRepository->findById($notificationId))) {
+            $this->error('Notification not found', ['notification_id' => $notificationId]);
+
+            return new NotFoundResponse('Notification');
+        }
+        $this->updatePropertiesInTransaction($request, $notification);
+
+        return new NoContentResponse();
+    }
+
+    /**
+     * @param PartialUpdateNotificationRequest $request
+     * @param Notification $notification
+     *
+     * @throws \Throwable
+     */
+    private function updatePropertiesInTransaction(
+        PartialUpdateNotificationRequest $request,
+        Notification $notification
+    ): void {
+        try {
+            $this->dataStorageEngine->startTransaction();
+            $this->updateIsActivated($request, $notification);
+            $this->dataStorageEngine->commitTransaction();
+        } catch (\Throwable $ex) {
+            $this->error('Rollback of \'PartialUpdateNotification\' transaction');
+            $this->dataStorageEngine->rollbackTransaction();
+
+            throw $ex;
+        }
+    }
+
+    /**
+     * @param PartialUpdateNotificationRequest $request
+     * @param Notification $notification
+     *
+     * @throws \Throwable
+     */
+    private function updateIsActivated(PartialUpdateNotificationRequest $request, Notification $notification): void
```

Comment:
method's name is misleading as you technicaly update the entire notification

Reply:
done